packages/core/lib/console.js

```diff
@@ -58,6 +64,26 @@ class Console extends EventEmitter {
     });
   }
 
+  // we only want to record a conflict the first time we try and load a contract
+  // into the repl context because we need to overwrite them when they change
+  recordNameConflicts(abstractions) {
```

Comment:
It seems this function detects new collisions and stores them opaquely. The object DS has to be mined to extract those new collisions, to alert the user. It might be better to maintain the collection of new collisions that can be directly used later on.